rust-lib/src/lib.rs

```diff
@@ -120,36 +125,64 @@ where
     }
 }
 
-#[derive(Debug, Deserialize)]
-struct StartRgbArgs {
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    network: bp::Chain,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    stash_endpoint: SocketLocator,
-    contract_endpoints: HashMap<ContractName, String>,
-    threaded: bool,
-    datadir: String,
+#[derive(Debug, Display, From, Error)]
+#[display(doc_comments)]
+#[non_exhaustive]
+enum RequestError {
+    /// String error
+    #[from]
+    StringErr(String),
+
+    /// Input value is not a JSON object
+    #[from]
+    SerdeErr(serde_json::Error),
```

Comment:
According to rust guidelines, we do not repeat `Error`/`Err` as a part of error name since the type itself contains it (i.e. `RequestError::Serde` is self-explanatory). I was adding `..Error` suffixes to enum variants all around before, so feel free to correct them if you will find anywhere.
```suggestion
    Json(serde_json::Error),
```

Reply:
good to know! thanks

and done